src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ProtectedPathSpec.groovy

```diff
@@ -59,34 +70,46 @@ Main and protected paths can't use the same link.""")
 
 class ProtectedPathSpec extends HealthCheckSpecification {
 
+    @Autowired
+    @Shared
+    FlowFactory flowFactory
+
+    @Autowired
+    @Shared
+    SwitchRulesFactory switchRulesFactory
+
     @Autowired @Shared
     Provider<TraffExamService> traffExamProvider
 
+    public static final Closure REQUIRED_COOKIE = { Long cookie ->  !new Cookie(cookie).serviceFlag && new Cookie(cookie).type == SERVICE_OR_FLOW_SEGMENT }
 
     @Tags(LOW_PRIORITY)
     def "Able to create a flow with protected path when maximumBandwidth=#bandwidth, vlan=#vlanId"() {
         given: "Two active not neighboring switches with two diverse paths at least"
         def switchPair = switchPairs.all().nonNeighbouring().withAtLeastNNonOverlappingPaths(2).random()
 
         when: "Create flow with protected path"
-        def flow = flowHelperV2.randomFlow(switchPair)
-        flow.allocateProtectedPath = true
-        flow.maximumBandwidth = bandwidth
-        flow.ignoreBandwidth = bandwidth == 0
-        flow.source.vlanId = vlanId
-        flowHelperV2.addFlow(flow)
+        def flow = flowFactory.getBuilder(switchPair)
+                .withProtectedPath(true)
+                .withBandwidth(bandwidth)
+                .withIgnoreBandwidth(bandwidth == 0)
+                .withSourceVlan(vlanId).build()
+                .create()
+
 
         then: "Flow is created with protected path"
-        def flowPathInfo = northbound.getFlowPath(flow.flowId)
-        flowPathInfo.protectedPath
+        def flowPathInfo = flow.retrieveAllEntityPaths()
+        !flowPathInfo.flowPath.protectedPath.isPathAbsent()
```

Comment:
Very minor, but what if we create a new method near the `isPathAbsent()`:
```
boolean isPathPresent() {
    !isPathAbsent()
}
```
This will simplify reading the test code

Reply:
let's use `!flowPathInfo.flowPath.protectedPath.isPathAbsent()`